src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,310 @@
+    total_score = 0
+    question_num = 0
+    for score in result.scores:
+        question_num += 1
+        print(f"Question #{question_num}:     {score['domain_specific_rubrics']}/5")
+        total_score += score["domain_specific_rubrics"]
+
+    average = total_score / len(result.scores)
+    average = round(average, 2)
+    print("----------------------------")
+    print(f"Average Score:   {average}/5")
+    print(f"Total Score:     {total_score}/{question_num*5}\n")
+
+
+def create_excel_results_file(excel_file: str, result: EvaluationResult):
+    summary_df = pd.DataFrame()
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    summary_df["scores"] = scores
+
+    question_indices = [f"Q{i + 1}" for i in range(len(summary_df))]
+    question_indices.append("Average")
+    question_indices.append("Total Score")
+    question_indices.append("Median")
+
+    average = summary_df.mean(axis=0, numeric_only=True)
+    total_score = summary_df.sum(axis=0, numeric_only=True)
+    median = summary_df.median(axis=0, numeric_only=True)
+
+    summary_df.loc[len(summary_df)] = average
+    summary_df.loc[len(summary_df)] = total_score
+    summary_df.loc[len(summary_df)] = median
+
+    summary_df.insert(0, "question index", question_indices)
+
+    response_df = result.dataset.to_pandas()
+    response_df["scores"] = scores
+
+    with pd.ExcelWriter(excel_file) as writer:
+        response_df.to_excel(writer, sheet_name="dataset", index=False)
+        summary_df.to_excel(writer, sheet_name="Summary", index=False)
+    return summary_df
+
+
+def write_results(
+    result: EvaluationResult, file_formats: str, output_dir: str, model_name: str
+):
+    response_df = result.dataset.to_pandas()
+    response_df["model_name"] = model_name
+
+    scores = [score["domain_specific_rubrics"] for score in result.scores]
+    response_df["scores"] = scores
+
+    now = datetime.now()
+    timestamp = now.strftime("%m-%d-%Y_%H-%M-%S-%f")
+    response_df["evaluation_run"] = f"run-{timestamp}"
+
+    print("Responses and scores written to:")
+    for fmt in file_formats:
+        if IOFileType.JSONL.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.JSONL.value, output_dir, timestamp, model_name
+            )
+            response_df.to_json(f"{results_file}", orient="records", lines=True)
+        elif IOFileType.CSV.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.CSV.value, output_dir, timestamp, model_name
+            )
+            response_df.to_csv(f"{results_file}", index=False)
+        elif IOFileType.XLSX.value == fmt:
+            results_file = create_results_file_name(
+                IOFileType.XLSX.value, output_dir, timestamp, model_name
+            )
+            create_excel_results_file(results_file, result)
+        else:
+            logger.debug("Output format %s is not valid", fmt)
```

Comment:
You're supporting QNA.yaml as a file type but it seems like you silently fail here.

Reply:
This function is just about output types and this initial PR is going to just have `.jsonl` as the input. I'll have another one open soon that supports `qna.yaml` as an input format and remove it from the list above in the mean time.

Reply:
I see. In this case I recommend having a sane default if we didn't validate their output format and they got this far. But I think we talked above that we should just validate as early as possible.